myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ExecutorCommandLineGenerator.java

```diff
@@ -19,11 +19,148 @@
+      //If SudoUser not enable LinuxExecutor will not work
+      appendSudo(cmdLine);
+      cmdLine.append("tar -zxpf ").append(getFileName(myriadExecutorConfiguration.getNodeManagerUri().get()));
+      cmdLine.append(" && ");
+      //Place the hadoop config where in the HADOOP_CONF_DIR where it will be read by the NodeManager
+      //The url for the resource manager config is: http(s)://hostname:port/yarnConfiguration so fetcher.cpp downloads the
+      //config file to yarnConfiguration, It's an xml file with the parameters of yarn-site.xml, core-site.xml and hdfs.xml.
+      if (!myriadExecutorConfiguration.getConfigUri().isPresent()) {
+        appendSudo(cmdLine);
+        cmdLine.append(" cp yarnConfiguration ");
+        cmdLine.append(myriadConfiguration.getYarnEnvironment().get("YARN_HOME"));
+        cmdLine.append("/etc/hadoop/yarn-site.xml && ");
+      }
+    }
+  }
+
+  protected void addJavaOpt(StringBuilder opts, String propertyName, String propertyValue) {
+    String envOpt = String.format(PROPERTY_FORMAT, propertyName, propertyValue);
+    opts.append(envOpt);
+  }
+
+  protected void appendSudo(StringBuilder cmdLine) {
+    if (myriadConfiguration.getFrameworkSuperUser().isPresent()) {
+      cmdLine.append(" sudo ");
+    }
+  }
+
+  protected void appendUserSudo(StringBuilder cmdLine) {
+    if (myriadConfiguration.getFrameworkSuperUser().isPresent()) {
+      cmdLine.append(" sudo -E -u ");
+      cmdLine.append(myriadConfiguration.getFrameworkUser().get());
+      cmdLine.append(" -H ");
+    }
+  }
+
+  public String getConfigurationUrl() {
+    String httpPolicy = yarnConfiguration.get(TaskFactory.YARN_HTTP_POLICY);
+    String address = StringUtils.EMPTY;
+    if (httpPolicy != null && httpPolicy.equals(TaskFactory.YARN_HTTP_POLICY_HTTPS_ONLY)) {
+      address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_WEBAPP_HTTPS_ADDRESS);
+      if (StringUtils.isEmpty(address)) {
+        address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_HOSTNAME) + ":8090";
+      }
+    } else {
+      address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_WEBAPP_ADDRESS);
+      if (StringUtils.isEmpty(address)) {
+        address = yarnConfiguration.get(TaskFactory.YARN_RESOURCEMANAGER_HOSTNAME) + ":8088";
+      }
+    }
+
+    return "http://" + address + "/yarnConfiguration";
+  }
+
+  private static String getFileName(String uri) {
+    int lastSlash = uri.lastIndexOf('/');
+    if (lastSlash == -1) {
+      return uri;
+    } else {
+      String fileName = uri.substring(lastSlash + 1);
+      Preconditions.checkArgument(!Strings.isNullOrEmpty(fileName), "URI should not have a slash at the end");
+      return fileName;
+    }
+  }
+
+  protected String getUser() {
+    if (myriadConfiguration.getFrameworkSuperUser().isPresent()) {
+      return myriadConfiguration.getFrameworkSuperUser().get();
+    } else {
+      return myriadConfiguration.getFrameworkUser().get();
+    }
+  }
```

Comment:
Another newbie question: if 1..n of these uris are not present, will there be bad state/thrown exceptions elsewhere?